department-of-veterans-affairs/caseflow
# tracks distribution task ready time, catches timer task job errors

Resolves #https://github.com/department-of-veterans-affairs/caseflow/issues/8755, https://github.com/department-of-veterans-affairs/caseflow/issues/8756

### Manual testing Plan

To test the locking behavior of the task timer job:

Run in console window 1:
```
timer = TaskTimer.find(1)
timer.update!(task_id: 666)
timer.with_lock { sleep 30; timer.update!(task_id: 888) }
```

Run in console window 2:
```
timer.with_lock { puts timer.task_id }
```
Observe that console window 2 hangs until console window 1 finishes, then prints `888`
